CRM/Financial/BAO/Payment.php

```diff
@@ -56,18 +56,17 @@ public static function create($params) {
     $contribution = civicrm_api3('Contribution', 'getsingle', ['id' => $params['contribution_id']]);
     $contributionStatus = CRM_Contribute_PseudoConstant::contributionStatus($contribution['contribution_status_id'], 'name');
 
+    $isPaymentCompletesContribution = self::isPaymentCompletesContribution($params['contribution_id'], $params['total_amount']);
```

Comment:
We should pass contribution's total_amount not $params because other then Pending, it is not true that $params[total_mount] is that of contribution's.

Reply:
Also, this function needs to be called at L82 immediate to ```recordPartialPayment``` that will decide wether the contribution actually completed.

Reply:
@monishdeb so my thinking is that function loads the contribution balance at the start of this FN & compares it against the amount being paid & if 0 or < 0 then the contribution is now fully paid (perhaps not the <0)

Reply:
Yup agree